src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -0,0 +1,2261 @@
+<?php
+/**
+ * Scans through an HTML document to find specific tags, then
+ * transforms those tags by adding, removing, or updating the
+ * values of the HTML attributes within that tag (opener).
+ *
+ * Does not fully parse HTML or _recurse_ into the HTML structure
+ * Instead this scans linearly through a document and only parses
+ * the HTML tag openers.
+ *
+ * ### Possible future direction for this module
+ *
+ *  - Prune the whitespace when removing classes/attributes: e.g. "a b c" -> "c" not " c".
+ *    This would increase the size of the changes for some operations but leave more
+ *    natural-looking output HTML.
+ *  - Decode HTML character references within class names when matching. E.g. match having
+ *    class `1<"2` needs to recognize `class="1&lt;&quot;2"`. Currently the Tag Processor
+ *    will fail to find the right tag if the class name is encoded as such.
+ *  - Properly decode HTML character references in `get_attribute()`. PHP's
+ *    `html_entity_decode()` is wrong in a couple ways: it doesn't account for the
+ *    no-ambiguous-ampersand rule, and it improperly handles the way semicolons may
+ *    or may not terminate a character reference.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ * @since 6.2.0
+ */
+
+/**
```

Comment:
Has this been tested against the doblock parser used fro the developer docs? https://github.com/WordPress/phpdoc-parser 

Also, needs a shorter summary for the developer docs title.

```suggestion
/**
 * Tokenize an HTML document.
 *
```

Reply:
Not to my knowledge. Are you able to help us by running that? I've never used it before so I can try and take a look, but I'm going to prioritize getting these review comments updated first and don't know if it's going to be easy or slow to figure out yet another new tool, or if I'll have some unexpected conflict on my system 🙃

Reply:
Shortened class summary in 4a438505ec